dbdialog/src/main/java/org/pentaho/ui/database/event/DataHandler.java

```diff
@@ -0,0 +1,1801 @@
+      }
+    }
+  }
+
+  private void setInfo( DatabaseMeta meta ) {
+
+    if ( meta == null ) {
+      return;
+    }
+
+    if ( meta.getAttributes().containsKey( EXTRA_OPTION_WEB_APPLICATION_NAME ) ) {
+      meta.setDBName( (String) meta.getAttributes().get( EXTRA_OPTION_WEB_APPLICATION_NAME ) );
+      meta.getAttributes().remove( EXTRA_OPTION_WEB_APPLICATION_NAME );
+      meta.setChanged();
+    }
+
+    getControls();
+
+    // Name:
+    if ( connectionNameBox != null ) {
+      connectionNameBox.setValue( meta.getDisplayName() );
+    }
+
+    PluginRegistry registry = PluginRegistry.getInstance();
+    PluginInterface dInterface = registry.getPlugin( DatabasePluginType.class, meta.getPluginId() );
+
+    // Connection type:
+    int index = ( dInterface == null ? -1 : new ArrayList<>( connectionMap.keySet() ).indexOf( dInterface.getName() ) );
+    if ( index >= 0 ) {
+      connectionBox.setSelectedIndex( index );
+    } else {
+      LogChannel.GENERAL.logError( "Unable to find database type "
+        + ( dInterface == null ? "null" : dInterface.getName() ) + " in our connection map" );
+    }
+
+    // Access type:
+    accessBox.setSelectedItem( DatabaseMeta.getAccessTypeDescLong( meta.getAccessType() ) );
+
+    // this is broken out so we can set the cache information only when caching
+    // connection values
+    setConnectionSpecificInfo( meta );
+    loadAccessData();
+
+    // Port number:
+    if ( portNumberBox != null ) {
+      portNumberBox.setValue( meta.getDatabasePortNumberString() );
+    }
+
+    // Options Parameters:
+
+    setOptionsData( meta.getExtraOptions() );
+
+    // Advanced panel settings:
+    if ( supportBooleanDataType != null ) {
+      supportBooleanDataType.setChecked( meta.supportsBooleanDataType() );
+    }
+
+    if ( supportTimestampDataType != null ) {
+      supportTimestampDataType.setChecked( meta.supportsTimestampDataType() );
+    }
+
+    if ( quoteIdentifiersCheck != null ) {
+      quoteIdentifiersCheck.setChecked( meta.isQuoteAllFields() );
+    }
+
+    if ( lowerCaseIdentifiersCheck != null ) {
+      lowerCaseIdentifiersCheck.setChecked( meta.isForcingIdentifiersToLowerCase() );
+    }
+
+    if ( upperCaseIdentifiersCheck != null ) {
+      upperCaseIdentifiersCheck.setChecked( meta.isForcingIdentifiersToUpperCase() );
+    }
+
+    if ( preserveReservedCaseCheck != null ) {
+      preserveReservedCaseCheck.setChecked( meta.preserveReservedCase() );
+    }
+
+    if ( strictBigNumberInterpretaion != null ) {
+      //check if Oracle
+      if ( meta.getDatabaseInterface() instanceof OracleDatabaseMeta ) {
+        strictBigNumberInterpretaion.setVisible( true );
+        strictBigNumberInterpretaion.setChecked( ( (OracleDatabaseMeta) meta.getDatabaseInterface() )
+          .strictBigNumberInterpretation() );
+      } else {
+        strictBigNumberInterpretaion.setVisible( false );
+        strictBigNumberInterpretaion.setChecked( false );
+      }
+    }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 921:931, 1523:1533.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.